.github/workflows/test-all.yml

```diff
@@ -200,8 +199,65 @@ jobs:
           PKG=${{ matrix.package }}
           gap -A --quitonbreak -r -c "SetInfoLevel(InfoPackageLoading, PACKAGE_WARNING);res:=TestPackage(\"$PKG\");FORCE_QUIT_GAP(res);"
 
+      - name: "Get job status"
+        id: job-status
+        if: ${{ always() }}
+        run: |
+          PKG="${{ matrix.package }}"
+          echo '::set-output name=status::${{ job.status }}'
+
       - name: "Run tests with OnlyNeeded"
+        if: ${{ always() }}
         timeout-minutes: 20
         run: |
           PKG=${{ matrix.package }}
           gap -A --quitonbreak -r -c "SetInfoLevel(InfoPackageLoading, PACKAGE_WARNING);res:=TestPackage(\"$PKG\":OnlyNeeded);FORCE_QUIT_GAP(res);"
+
+      - name: "Get job status with OnlyNeeded"
+        id: job-status-only-needed
+        if: ${{ always() }}
+        run: |
+          PKG="${{ matrix.package }}"
+          echo '::set-output name=status::${{ job.status }}'
+
+      - name: "Create job status json-file"
+        if: ${{ always() }}
+        run: |
+          PKG="${{ matrix.package }}"
+          STATUS="${{ steps.job-status.outputs.status }}"
+          STATUS_ONLY_NEEDED="${{ steps.job-status-only-needed.outputs.status }}"
+          jq --null-input \
+            --arg status "${STATUS}" \
+            --arg status_only_needed "${STATUS_ONLY_NEEDED}" \
+            '{"status": $status, "status_only_needed": $status_only_needed}' > ${PKG}.json
+          cat ${PKG}.json
+
+      - name: "Upload job status as artifact"
+        if: ${{ always() }}
+        uses: actions/upload-artifact@v2
+        with:
+          name: "reports-${{ matrix.package }}"
+          path: "${{ matrix.package }}.json"
+
+  report:
+    name: "Report"
+    needs: test-package
+    if: ${{ always() }}
+    runs-on: ubuntu-18.04
+    steps:
+      - name: "Create _reports directory"
+        run: |
+          mkdir -p _reports
+
+      - name: "Download every job status"
+        uses: elonh/download-artifact-regexp@master # FIXME/TODO: Switch to actions/download-artifact once they officially support wildcards, see https://github.com/actions/download-artifact/issues/6
+        with:
+         pattern: reports-*
+         path: _reports
+
+      - name: "Show Files"
+        run: |
+          cd _reports
+          ls
+          cd reports-ace
+          ls
```

Comment:
So this bit should be turned into an output of the workflow, along the lines of https://docs.github.com/en/actions/using-workflows/reusing-workflows#using-outputs-from-a-reusable-workflow

Then we'll invoke `test-all.yml` from various other workflows which can handle the output as they deem fit

Reply:
This makes sense. So at this stage we would just collect all data into `test-status.json` with some additional information and turn this into the output. I probably would write a python script for this, so we could test locally with custom stuff in `_reports`.

Reply:
Ok, I turned this now into an output for workflow calls.